client/ayon_houdini/plugins/publish/collect_arnold_rop.py

```diff
@@ -70,20 +70,31 @@ def process(self, instance):
         multipartExr = True
 
         num_aovs = rop.evalParm("ar_aovs")
-        # TODO: Check the following logic.
-        #       as it always assumes that all AOV are not merged.
+
         for index in range(1, num_aovs + 1):
-            # Skip disabled AOVs
-            if not rop.evalParm("ar_enable_aov{}".format(index)):
+            
+            aov_enabled = rop.evalParm("ar_enable_aov{}".format(index)) 
+            aov_sep = rop.evalParm("ar_aov_separate{}".format(index))
+            aov_path = rop.evalParm("ar_aov_separate_file{}".format(index))
+            
+            # Skip disabled AOVs or AOVs with no separate aov file path
+            if not all((aov_enabled, aov_path, aov_sep)):
                 continue
 
             if rop.evalParm("ar_aov_exr_enable_layer_name{}".format(index)):
                 label = rop.evalParm("ar_aov_exr_layer_name{}".format(index))
             else:
                 label = evalParmNoFrame(rop, "ar_aov_label{}".format(index))
 
-            aov_product = self.get_render_product_name(default_prefix,
-                                                       suffix=label)
+            # NOTE:
+            #  we don't collect the actual AOV path but rather assume 
+            #    the user has used the default beauty path (collected above)
+            #    with the AOV name before the extension.
+            #  Also, Note that Ayon Publishing does not require a specific file name,
+            #    as it will be renamed according to the naming conventions set in the publish template.
```

Comment:
Sorry - why do we assume that? Isn't it trivial to 'fix' that?

Also it seems that all [this logic](https://github.com/ynput/ayon-houdini/blob/a7c1caa098a7e623956023420366687a44a3c2f6/client/ayon_houdini/plugins/publish/collect_arnold_rop.py#L133-L145) isn't doing anything at all for Arnold. `${AOV}` and `$AOV` and `%AOV%` do nothing with the Arnold ROP. So that logic can be removed? Or?

Because we're now also due to this failing to correctly detect the images if they just happen to have a very different path set?

Reply:
We should somehow differentiate the logic between:

- What are the expected file paths? (We know that, because that's explicitly defined on the parm, yay!)
- What will we make the product name for the AOV - preferably that's the main product name, but with a suffix of the AOV or alike? So we'll somehow need to find a part of the filename the user has defined that we can use for that.

Reply:
> Sorry - why do we assume that? Isn't it trivial to 'fix' that?
> 
> Also it seems that all [this logic](https://github.com/ynput/ayon-houdini/blob/a7c1caa098a7e623956023420366687a44a3c2f6/client/ayon_houdini/plugins/publish/collect_arnold_rop.py#L133-L145) isn't doing anything at all for Arnold. `${AOV}` and `$AOV` and `%AOV%` do nothing with the Arnold ROP. So that logic can be removed? Or?
> 
> Because we're now also due to this failing to correctly detect the images if they just happen to have a very different path set?

We need to improve (and even remove) the logic above to make sure the AOV files included in the expected file paths. 
And we need to think about how to deal with the cryptomatte too: https://help.autodesk.com/view/ARNOL/ENU/?guid=arnold_for_houdini_shaders_ah_Cryptomatte_html

Reply:
So this PR fixes a current critical issue already - and the 'naming' issue here isn't new and was already present. I'd say, let's merge this now and keep the comment and create a new issue @MustafaJafar to track for us that we should fix this naming for the separate layers down the line.